### ISSUE TITLE
Daemon Crash: "Couldn't unregister mask with UUID <UUID> from parent <UUID>, mask not registered with that parent"

### ISSUE THREAD

**Issue by yourbuddyconner** [CodaProtocol/coda#3032]
The coda-services daemon crashed with another new error, looks like it crashed immediately following several disconnects from peers. 

```
2019-07-27 22:26:42 UTC [Error] Broadcasting message "new state" to $peer failed: $error
peer: {
"host": "213.230.114.93",
"discovery_port": 53247,
"communication_port": 53246
}
error: "(Unix.Unix_error \"Connection timed out\" connect 213.230.114.93:53246)"
2019-07-27 22:26:42 UTC [Error] Broadcasting message "new state" to $peer failed: $error
peer: {
"host": "213.230.114.93",
"discovery_port": 53247,
"communication_port": 53246
}
error: "(Unix.Unix_error \"Connection timed out\" connect 213.230.114.93:53246)"
2019-07-27 22:27:44 UTC [Info] Some peers disconnected [host : 213.230.114.93, discovery_port : 53247, communication_port : 53246]
2019-07-27 22:27:44 UTC [Info] Removing peer from peer set: [host : 213.230.114.93, discovery_port : 53247, communication_port : 53246]
2019-07-27 22:28:19 UTC [Error] RPC call raised an exception: $exn
exn: "(Unix.Unix_error \"Connection timed out\" connect 213.230.114.93:53246)"
2019-07-27 22:28:19 UTC [Error] Broadcasting message "snark pool diff" to $peer failed: $error
peer: {
"host": "213.230.114.93",
"discovery_port": 53247,
"communication_port": 53246
}
error: "(Unix.Unix_error \"Connection timed out\" connect 213.230.114.93:53246)"
☠ Coda Daemon crashed. The Coda Protocol developers would like to know why!
Please:
Open an issue:
https://github.com/CodaProtocol/coda/issues/new
Briefly describe what you were doing, and include the last 20 lines from .coda-config/coda.log.
And then paste the following:
(monitor.ml.Error
(Failure
"Couldn't unregister mask with UUID e22e0403-3805-368b-4d2d-1a2864dddb63 from parent 1c76fee7-7e1b-32e4-cdbd-ca021b0843e1, mask not registered with that parent")
("Raised at file \"stdlib.ml\", line 33, characters 17-33"
"Called from file \"lib/merkle_mask/maskable_merkle_tree.ml\", line 159, characters 12-72"
"Called from file \"lib/coda_base/ledger.ml\" (inlined), line 167, characters 35-79"
"Called from file \"lib/transition_frontier/transition_frontier.ml\", line 784, characters 24-63"
"Called from file \"lib/transition_frontier/transition_frontier.ml\", line 787, characters 14-35"
"Called from file \"list.ml\", line 106, characters 12-15"
"Called from file \"src/list0.ml\" (inlined), line 26, characters 40-75"
"Called from file \"lib/transition_frontier/transition_frontier.ml\", line 803, characters 12-66"
"Called from file \"lib/transition_handler/processor.ml\", line 64, characters 17-51"
"Called from file \"lib/transition_handler/processor.ml\", line 149, characters 8-129"
"Called from file \"src/deferred0.ml\", line 56, characters 64-69"
"Called from file \"src/job_queue.ml\" (inlined), line 131, characters 2-5"
"Called from file \"src/job_queue.ml\", line 170, characters 6-47"
"Caught by monitor coda"))
```

**Comment by enolan** [comment]
Let's see if this is fixed after #3009 lands.

**Comment by o1pranay** [comment]
Closing this one out - dupe of #2969
